static/context.html

```diff
@@ -27,7 +27,15 @@
   <!-- Dynamically replaced with <script> tags -->
   %SCRIPTS%
   <script type="text/javascript">
-    window.__karma__.loaded();
+    if (window.addEventListener) {  
+      document.addEventListener("DOMContentLoaded",function(){
+        window.__karma__.loaded();
```

Comment:
I think we should use "Load" here rather than "DOMContentLoaded".  Any content in any <link> tag or even <img> could be needed by a test.

Reply:
Wouldn't this change the actual behavior, where tests are run without waiting for all content to be loaded?

Reply:
it might be better to make this an option rather than changing the behavior for everyone where it worked well so far

Reply:
What could be an option? the type of event (loaded vs DomContentLoaded) or the whole new stuff?

Reply:
In existing code, the `__karma__.loaded()` is called after all the script tags are added to the document, at then end of the document.  Adding an event listener will cause an extra delay, the .loaded() function will run in a different event-loop turn, and the stylesheet/css content will be loaded. Any of these three could cause some marginal tests to fail, but such tests are working outside of the expectations of the system.  Tests that rely on time between outer-code load and test start, on synchrony with the outer-code load, or on the non-existence of resources lie outside of what karma-runner was designed to support IMO.  

So the real question is whether this affects any of the popular test frameworks, which could have dependency on how `loaded()` is called.   I guess the most likely issue would be the extra event-loop turn.  I can't see how loaded vs domcontentloaded could make a difference except for cases with tight timeouts.

Reply:
I agree with @johnjbarton analysis, and my conclusion of this analysis is that waiting for DomContentLoaded or for Loaded event is not a breaking change.

What is not sure for me is why you would want to use this pull request to wait longer, for "loaded" event? I think that you are right, it is a small delay more, out of the boundaries of the expected, but why do you want that more than just what is necessary for my bug fix?

Reply:
Because the issue that you have -- failure of tests because an asynchronously loaded resource is missing -- is the same problem we have for other resources.  For many, many tests that have nothing other than JS, loaded vs documentcontentloaded should make no difference. Only if we have resources we want will it make a difference, and that difference will be positive.

Reply:
I propose to make this PR pass with the minimum change in behavior, and to open another ticket for the behavior you describe. If this PR pass, it will be very easy for you to make this new PR.